docs/kepler_model_server/node_profile.md

```diff
@@ -0,0 +1,7 @@
+# Node Profile
+
+To form a group of machines (nodes), the idea is to run a benchmark suite towards a bunch of machines, collect performance reported by the benchmarks, and apply clustering algorithm such as *kmeans*.
+
+For each group of [node type](./pipeline.md#node-spec), we make a profile composing of background power when the resource usage is almost constant without user workload, minimum, maximum power for each power components (e.g., core, uncore, dram, package, platform), and normalization scaler (i.e., MinMaxScaler), standardization scaler (i.e., StandardScaler) for each [feature group](./pipeline.md#available-metrics).
```

Comment:
Can you provide more details about the normalization scaler and standardization scaler? Maybe include the formulas and why we use them here.

Also, how they are they selected?

Reply:
The Kepler code has the Liner Regression normalization as:
```
normalizedX := (vals[index] - coeff.Mean) / math.Sqrt(coeff.Variance)
```
However, as I mentioned before the Mean can be a very high value and then the low utilization will return a negative value, which means a negative power consumption. Therefore, we must change this normalization.

Reply:
It uses the standard scaler from [scikit learn](http://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.StandardScaler.html) library.  Specifically, it uses [SGDRegressor](https://github.com/sustainable-computing-io/kepler-model-server/blob/main/src/train/trainer/SGDRegressorTrainer/main.py) to generate the weight.

Reply:
> class sklearn.preprocessing.StandardScaler(*, copy=True, with_mean=True, with_std=True)
> Standardize features by removing the mean and scaling to unit variance.
> 
> The standard score of a sample x is calculated as:
> 
> z = (x - u) / s
> 
> where u is the mean of the training samples or zero if with_mean=False, and s is the standard deviation of the training samples or one if with_std=False.

@sunya-ch So, we can set `with_mean=False` to avoid the problem that I mentined.

Reply:
MinMaxScaler looks to see the right approach.